models/branches.go

```diff
@@ -111,12 +127,28 @@ func (protectBranch *ProtectedBranch) GetGrantedApprovalsCount(pr *PullRequest)
 		return 0
 	}
 
+	repo, err := GetRepositoryByID(protectBranch.RepoID)
+	if err != nil {
+		log.Error("GetRepositoryByID: %v", err)
+		return 0
+	}
+
 	approvals := int64(0)
 	userIDs := make([]int64, 0)
 	for _, review := range reviews {
 		if review.Type != ReviewTypeApprove {
 			continue
 		}
+		if !protectBranch.EnableApprovalsWhitelist {
```

Comment:
IMHO approvals should "remember" if the users were whitelisted at the moment they approved and this value should not be recalculated. Two reasons for this:

* This is not "list friendly"; it's very difficult to optimize as we want to have the PR list showing the number of approvals. If possible, we should resolve the list with a simple join/select.
* If rules change, history should not be rewritten. Non-counting approvals should remain non-counting and vice-versa.

Reply:
Your argumentation makes sense, but that would require quite large code change (need new database columns). I think it should be a different PR.

Reply:
I don't think it's that big of a change. I think you only need a new "Verified bool"  column to the review table and move this check to the code that inserts the review. One day that "verified" column can reflect a 2FA, for example.

Reply:
Can it be called `OfficialReviewer  (bool)`? 
You mean there will be an option to only accept 2FA users as reviewers?

Reply:
Yes, it can be called OfficialReviewer, although that would lock the meaning. What I mean is that what today is a bool (verified/not verified) tomorrow can be an enum (not verified/verified/verified+signed, etc.). But whatever we may want in the future can be solved with a migration no problem, so feel free to choose whatever you feel that fits best. 👍